google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/OptionsTest.java

```diff
@@ -72,7 +72,6 @@ public void allOptionsAbsent() {
     assertThat(options.toString()).isEqualTo("");
     assertThat(options.equals(options)).isTrue();
     assertThat(options.equals(null)).isFalse();
-    assertThat(options.equals(this)).isFalse();
```

Comment:
This seems like an interesting test.  Why was it removed?

Reply:
this is testing 2 different objects( OptionsTest against Options). This may not be interesting because it should always result in false.

Reply:
Testing equals against different objects seems like it would exercise a unique codepath, which is "interesting".  Shouldn't we keep this test?